scripts/tests

```diff
@@ -25,90 +25,85 @@ def sha256sum(result):
         s = ""
         for k, v in dict.items():
             s += f"{k}={v} "
         return s
 
-    if force_qemu or (platform.system() == "Linux" and platform.machine() == "x86_64"):
-        logging.debug(f"Emulating {bin} with QEMU")
+    if platform.system() == "Linux" and platform.machine() == "x86_64":
+        logging.debug(f"Cross compiling {bin}")
 
         args = [
             "make",
             "CROSS_PREFIX=aarch64-none-linux-gnu-",
             f"{bin}",
         ] + extra_make_args
 
-        logging.info(dict2str(extra_make_envs) + " ".join(args))
-
-        p = subprocess.run(
-            args,
-            stdout=subprocess.DEVNULL if not verbose else None,
-            env=os.environ.copy() | extra_make_envs,
-        )
-        if p.returncode != 0:
-            logging.error(f"make failed: {p.returncode}")
-            sys.exit(1)
-
-        result = subprocess.run(
-            ["qemu-aarch64", f"{bin}"],
-            capture_output=True,
-            universal_newlines=False,
-        )
-
-        if result.returncode != 0:
-            logging.error(
-                f"Emulating {bin} failed: {result.returncode} {result.stderr.decode()}"
-            )
-            sys.exit(1)
-
     else:
-        logging.debug(f"Running {bin} natively")
+        logging.debug(f"Compiling {bin} natively")
 
         args = ["make", f"{bin}"] + extra_make_args
         logging.info(dict2str(extra_make_envs) + " ".join(args))
 
-        p = subprocess.run(
-            args,
-            stdout=subprocess.DEVNULL if not verbose else None,
-            env=os.environ.copy() | extra_make_envs,
-        )
+    p = subprocess.run(
+        args,
+        stdout=subprocess.DEVNULL if not verbose else None,
+        env=os.environ.copy() | extra_make_envs,
+    )
+
+    if p.returncode != 0:
+        logging.error(f"make failed: {p.returncode}")
+        sys.exit(1)
 
-        if p.returncode != 0:
-            logging.error(f"make failed: {p.returncode}")
-            sys.exit(1)
 
-        cmd = [f"./{bin}"]
+def base_run(
+    bin,
+    force_qemu,
+    verbose,
+    run_as_root=False,
+    exec_wrapper=None,
+    mac_taskpolicy=None,
+):
+    """Run the binary in all different ways"""
+    cmd = [f"./{bin}"]
+    if force_qemu or (platform.system() == "Linux" and platform.machine() == "x86_64"):
+        logging.info(f"Emulating {bin} with QEMU")
+        cmd = ["qemu-aarch64"] + cmd
+    else:
+        if exec_wrapper:
+            logging.info(f"Running {bin} with customized wrapper.")
+            cmd = [f"{exec_wrapper}"] + cmd
+
         if run_as_root:
             logging.info(
-                "Running benchmarks as root -- you may need to enter your root password."
+                f"Running {bin} as root -- you may need to enter your root password."
             )
             cmd = ["sudo"] + cmd
 
-        if mac_taskpolicy is not None:
+        if platform.system() == "Darwin" and mac_taskpolicy is not None:
```

Comment:
Can we subsume `mac_taskpolicy` under `exec_wrapper`? That is, if `mac_taskpolicy` is defined, set `exec_wrapper` accordingly. If both `mac_taskpolicy` and `exec_wrapper` are defined, fail. Document accordingly that `mac_taskpolicy` is just an abbreviation for `exec_wrapper {taskpolicy ...}`.